src/data/viteComboFetcher.ts

```diff
@@ -0,0 +1,134 @@
+  const asObject = YamlComboTableSchemaZod.parse(parse(await readFile(path, "utf8")))
+
+  if(!asObject.tricks) {
+    throw new Error("Combos must contain at least one trick.")
+  }
+
+  asObject.tricks = asObject.tricks.map( trickId => ([trickId, "official"])) as any
+
+  // Makes sure the YAML has the right structure. 
+  return DbCombosTableZod.parse({...asObject, comboStatus: "official"})
+}
+
+/**
+ * This takes a list of Database Entities and creates a mapping of ID → List of Objects
+ * 
+ * This can be used to detect duplicates or to check if an Entity with a given ID exists.
+ */
+function createRecordLookup<T extends z.infer<typeof DbCombosTableZod> | z.infer<typeof DbTricksTableZod>>(allEntities: T[]) {
+  let idToEntityLookup: Record<number, T[]> = {};
+
+  allEntities.forEach(databaseEntity => {
+    if (!idToEntityLookup[databaseEntity.id]) {
+      idToEntityLookup[databaseEntity.id] = [];
+    }
+      idToEntityLookup[databaseEntity.id].push(databaseEntity);
+    });
+  return idToEntityLookup;
+}
+
+function findDuplicateKeys(allCombos: z.infer<typeof DbCombosTableZod>[]) {
+
+  let idToTrickLookup: Record<number, (z.infer<typeof DbCombosTableZod>)[]> = createRecordLookup(allCombos);
+
+  return Object.entries(idToTrickLookup).filter(([_,trick]) => trick.length > 1).map(([k,v]) => ({
+    id: k,
+    technicalNames: v.map( trick => trick.technicalName)
+  }))
+}
+
+function findUndefinedTrickReferences(allCombos: z.infer<typeof DbCombosTableZod>[], allTricks: z.infer<typeof DbTricksTableZod>[]) {
+  
+  let idToTrickLookup: Record<number, (z.infer<typeof DbTricksTableZod>)[]> = createRecordLookup(allTricks);
+
+  // issues define the ID of the Combo and an issue message
+  const issues: [number, string][] = []
+
+  allCombos.forEach( combo => {
+    if(combo.tricks) {
+      combo.tricks.forEach( (f,i) => {
+        if(f[1] !== "official") {
+          issues.push([combo.id, `All tricks must limit themselves to official ones (not the case for ${i+1}. Entry)`])
+        }
+        else if(!idToTrickLookup[f[0]]) {
+          issues.push([combo.id, `${i+1}. Entry in tricks references an ID that does not exist (id: ${f[0]}).`])
+        }
+
+      })
+    }
+  })
+
+  return issues;
+}
+
+
+
+
+
+export default async function viteGetAllCombos(tricks: z.infer<typeof DbTricksTableZod>[]) {
+  // @ts-expect-error Because there are effectively two TS Projects (Vite Plugin Context 
+  //and Vue Webapp Context), TS gets a bit confused and doesn't think import.meta.url is allowed here
+  const pattern = join(fileURLToPath(import.meta.url), "..", "combos", "*.yaml");
+  const allFilePaths = await globby(pattern)
+
+  const parsedYamlFiles = await Promise.allSettled(allFilePaths.map( comboFilePath => fetchYamlFile(comboFilePath)))
+  const erroredFiles = parsedYamlFiles.map( (fileResult,i) => fileResult.status === "rejected" ? ({error: fileResult.reason, path: allFilePaths[i]}) : (undefined as never) ).filter(Boolean);
```

Comment:
What does `.filter(Boolean)` do?

Reply:
`.filter` accepts a function as its argument. If that function returns thruthy, that entry will be found in the returned array. It its falsy, it wont be found there.

You will usually see something like `array.filter( e => e > 42)`. `e => e > 42` is a function declaration. `Boolean` is the Boolean Constructor. If we use the lambda-Notation, it would be equivalent to `array.filter( e => Boolean(e))`. So it turns e into a true or false value. Do note however that, because the result does not have to be `true` or `false`, but only `truthy` or `falsy`, `array.filter(Boolean)` has the exact same effect as `array.filter( e => e)`

Reply:
So in Effect it doesnt really matter, but from what I've heard doing a `filter(Boolean)` will tell TypeScript in the future that the result cannot have undefined's and null's.

So this would not throw a type error:
```ts

const arr: (string | undefined)[] = ["asdf", undefined, "bla"]
               //  vvvvv- this will currently emit a TS error
const arrNoUndef: string[] = arr.filter(Boolean)
```

Reply:
Interesting!
Thanks for the clarification :)